message-queue-scraper/__init__.py

```diff
@@ -0,0 +1,108 @@
+    message_dict = json.loads(queue_message)
+    case_urls = message_dict["case-urls"]
+    base_url = message_dict["scrape-params"]["base-url"]
+    county = message_dict["scrape-params"]["county"]
+    odyssey_version = message_dict["scrape-params"]["odyssey-version"]
+    search_url = message_dict["scrape-params"]["search-url"]
+    notes = message_dict["scrape-params"]["notes"]
+    date_string = message_dict["scrape-params"]["date-string"]
+    date_string_underscore = date_string.replace("/", "_")
+    JO_id = message_dict["scrape-params"]["JO-id"]
+    # is this always going to work?
+    hidden_values = message_dict["scrape-params"]["hidden-values"]
+    # # TODO - get these from message
+    # ms_wait = 200
+    # location = None
+    ms_wait = message_dict["scrape-params"]["ms-wait"]
+    location = message_dict["scrape-params"]["location"]
+    court_calendar_link_text = "Court Calendar"
+
+    # initialize session
+    session = requests.Session()
+    # allow bad ssl and turn off warnings
+    session.verify = False
+    requests.packages.urllib3.disable_warnings(
+        requests.packages.urllib3.exceptions.InsecureRequestWarning
+    )
+
+    # initialize blob container client for sending html files to
+    blob_connection_str = os.getenv("AzureWebJobsStorage")
+    container_name_html = os.getenv("blob_container_name_html")
+    blob_service_client: BlobServiceClient = BlobServiceClient.from_connection_string(
+        blob_connection_str
+    )
+    container_client = blob_service_client.get_container_client(container_name_html)
+
+    # The following 2 searches' results are not actually used in this function -- only here because 
+    # this search is a necessary preliminary step to get the case URLs to load.
+    # hit the search page to gather initial data
+    search_page_html = request_page_with_retry(
+        session=session,
+        url=search_url
+        if odyssey_version < 2017
+        else urllib.parse.urljoin(base_url, "Home/Dashboard/26"),
+        verification_text="Court Calendar"
+        if odyssey_version < 2017
+        else "SearchCriteria.SelectedCourt",
+        http_method=HTTPMethod.GET,
+        ms_wait=ms_wait,
+    )
+
+    # POST a request for search results
+    results_page_html = request_page_with_retry(
+        session=session,
+        url=search_url
+        if odyssey_version < 2017
+        else urllib.parse.urljoin(
+            base_url, "Hearing/SearchHearings/HearingSearch"
+        ),
+        verification_text="Record Count"
+        if odyssey_version < 2017
+        else "Search Results",
+        data=create_search_form_data(
+            date_string, JO_id, hidden_values, odyssey_version
+        ),
+        ms_wait=ms_wait,
+    )
+
+    # actual scraping of case urls passed in message
+    for case_url in case_urls:
+        case_id = case_url.split("=")[1]

```

Comment:
NIce, this is easier than I thought it would be (my assumption is we would have to initiate a search result that would return that specific case